doc/stake-pool-operations/KES_period.md

```diff
@@ -4,43 +4,185 @@ To create an operational certificate for a block-producing node, you need a _KES
 
 Here "KES" stands for _**K**ey **E**volving **S**ignature_, which means that after a certain _period_, the key will _evolve_ to a new key
 and discard its old version. This is useful, because it means that even if an attacker compromises the key and gets access to the signing key,
-he can only use that to sign blocks _from now on_, but not blocks dating from _earlier periods_, making it impossible for the attacker to rewrite history.
+he can only use that to sign blocks _from now on_, but not blocks dating from _earlier periods_, making it impossible for the attacker to rewrite history. To generate a new KES key pair, see below.
 
 A KES key can only evolve for a certain number of periods and becomes useless afterwards.
 This means that before that number of periods has passed, the node operator has to generate a new KES key pair, issue a new operational node certificate with that new key pair and restart the node with the new certificate.
 
+### KES key generation
+
+```bash
+cardano-cli node key-gen-KES \
+--verification-key-file kes.vkey \
+--signing-key-file kes.skey
+```
+
 To find out how long one period is and for how long a key can evolve, we can look into the _genesis file_. If that file is called `mainnet-shelley-genesis.json`,
-we can type
+we can type:
 
-    cat mainnet-shelley-genesis.json | grep KES
-    "slotsPerKESPeriod": 129600,
-    "maxKESEvolutions": 62,
+```bash
+cat mainnet-shelley-genesis.json | grep KES
+"slotsPerKESPeriod": 129600,
+"maxKESEvolutions": 62,
+```
 
-in this example, the key will evolve after each period of 129600 slots and that it can evolve 62 times before it needs to be renewed.
+The key will evolve after each period of 129600 slots and it can evolve up to 62 times before it needs to be renewed.
 
 Before we can create an operational certificate for our node, we need to figure out the start of the KES validity period, i.e. which KES evolution period we are in.
 
 We check the current tip of the blockchain:
 
-    cardano-cli query tip --mainnet
+```bash
+cardano-cli query tip --mainnet
+{
+    "epoch": 259,
+    "hash": "dbf5104ab91a7a0b405353ad31760b52b2703098ec17185bdd7ff1800bb61aca",
+    "slot": 26633911,
+    "block": 5580350
+}
+```
+In this example, we are currently in slot 26633911, and we know from the genesis file that one period lasts for 129600 slots. So we calculate the current period:
+
+```bash
+expr 26633911 / 129600
+> 205
+```
+
+### Issue operational certificate
+```bash
+cardano-cli node issue-op-cert \
+--kes-verification-key-file kes.vkey \
+--cold-signing-key-file cold.skey \
+--operational-certificate-issue-counter cold.counter \
+--kes-period 205 \
+--out-file opcert.cert
+```
+**This command increments the cold counter by 1.**
+
+## :warning: Vasil hard fork breaking changes
+
+Note that from the Vasil hard fork onwards there is a new rule for the cold.counter:
```

Comment:
"for the operational certificate's counter". Out of curiosity why is it being referred to as a cold counter?

Reply:
comes from the naming used here -->
https://github.com/input-output-hk/cardano-node/blame/master/doc/stake-pool-operations/node_keys.md